### PR TITLE
Верховых Михаил Дмитриевич 2537

### PR DESCRIPTION
**Pull request by mihver1**
Commit history is available: github.com/mihver1/extra

### CODE REVIEW

**Comment by thevery** [comment]
top/bottom padding is missing

**Comment by mihver1** [comment]
Nothing was said about top/bottom padding in the task.

**Comment by thevery** [comment]
Yes. That's what we call "common sense" in our tasks' special condition "Ничего из условия выше не заменит здравого смысла."

**Comment by thevery** [comment]
placeholder for photo is nice, though images are jumping because of that

**Comment by thevery** [comment]
no action bar on the second screen is good, but the photos are not full size

**Comment by thevery** [comment]
you should not reload photo on screen rotate

**Comment by thevery** [comment]
offline mode is definitely broken - only small images are cached

**Comment by thevery** [comment]
it is better to put update button to action bar (and you can put some branding to action bar as well!)

**Comment by thevery** [comment]
app is ok (20-5=15), code review pending

File: src/ru/mihver1/android/yaph/IconCache.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            result = loadBitmap(url);
+            if (cache.putIfAbsent(url, result) != null) {
+                result.recycle();
+            }
+        }
+        return getBitmapFromCache(url);
+    }
+
+    private Bitmap loadBitmap(String url) throws IOException {
+        Lock lock = getLock(url);
```
Comment (thevery) [inline src/ru/mihver1/android/yaph/IconCache.java:48]: why do you need lock here?

File: src/ru/mihver1/android/yaph/IconCache.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        lock.lock();
+        Log.d("IconCache", "start " + url);
+        try {
+            Bitmap bt = BitmapFactory.decodeStream(new URL(url).openStream());
+            String root = Environment.getExternalStorageDirectory().toString();
+            File myDir = new File(root + "/flickr_cache");
+            myDir.mkdirs();
+            String fname = url.substring(url.length() - 9);
```
Comment (thevery) [inline src/ru/mihver1/android/yaph/IconCache.java:56]: why 9?

File: src/ru/mihver1/android/yaph/ImagePage.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+import java.io.IOException;
+import java.net.URL;
+
+/**
+ * Created by mihver1 on 19.01.14.
+ */
+public class ImagePage extends Activity {
+
+    public static int W, H;
```
Comment (thevery) [inline src/ru/mihver1/android/yaph/ImagePage.java:24]: 1 line per declaration

File: src/ru/mihver1/android/yaph/ImagePage.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        Display display = getWindowManager().getDefaultDisplay();
+        Point size = new Point();
+        display.getSize(size);
+        W = size.x;
+        H = size.y;
+
+        Bundle bundle = getIntent().getExtras();
+
+        ImageView fullscreen = (ImageView) findViewById(R.id.imageView2);
+        fullscreen.setTag(bundle.getString("image"));
```
Comment (thevery) [inline src/ru/mihver1/android/yaph/ImagePage.java:46]: `"image"` should be const

File: src/ru/mihver1/android/yaph/ImagePage.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        fullscreen.setImageResource(R.drawable.placeholder);
+
+        Log.d("YOLO", bundle.getString("image"));
```
Comment (thevery) [inline src/ru/mihver1/android/yaph/ImagePage.java:49]: bad logs

File: src/ru/mihver1/android/yaph/ImagePage.java [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+                h = temp.getHeight();
+                double ratio = 1.0f;
+                if(w > W || h > H) {
+                    ratio = Math.min(W*1.0 / w, H*1.0 / h);
+                    Log.d("YOLO", ">>"+Integer.toString((int)(w * ratio))+" "+Integer.toString((int)(h * ratio)));
+                }
+                return Bitmap.createScaledBitmap(temp, (int)(w * ratio), (int)(h * ratio), false);
+            } catch (IOException e) {
+                return null;
+            } catch (NullPointerException e) {
```
Comment (thevery) [inline src/ru/mihver1/android/yaph/ImagePage.java:83]: catch NPE, really?

File: src/ru/mihver1/android/yaph/MainPage.java [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+
+public class MainPage extends Activity {
+    /**
+     * Called when the activity is first created.
+     */
+
+    private String FLICKR_API_KEY;
+
+    // http://stackoverflow.com/questions/1560788/how-to-check-internet-access-on-android-inetaddress-never-timeouts
+    private boolean isOnline() {
```
Comment (thevery) [inline src/ru/mihver1/android/yaph/MainPage.java:46]: code duplication

File: src/ru/mihver1/android/yaph/MainPage.java [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+
+        @Override
+        protected void onPostExecute(ArrayList<String> strings) {
+            Log.d("YOLO", "onPostExecute");
+            urls = strings;
+            Log.d("YOLO", "Size " + Integer.toString(strings.size()));
+
+            ListView listView = (ListView) findViewById(R.id.listView);
+
+            if(getResources().getConfiguration().orientation == Configuration.ORIENTATION_PORTRAIT) {
```
Comment (thevery) [inline src/ru/mihver1/android/yaph/MainPage.java:130]: use `values` and `values-land` resource folders instead of 2 copy-paste adapters

**Comment by thevery** [comment]
20-5-3=12

File: src/ru/mihver1/android/yaph/IconCache.java (hunk 2 above)
Comment (mihver1) [inline src/ru/mihver1/android/yaph/IconCache.java:56]: Why not? It works.

File: src/ru/mihver1/android/yaph/ImagePage.java (hunk 5 above)
Comment (mihver1) [inline src/ru/mihver1/android/yaph/ImagePage.java:49]: Common user won't read logs.
This log was helpful for me. Why is it bad?

File: src/ru/mihver1/android/yaph/IconCache.java (hunk 2 above)
Comment (thevery) [inline src/ru/mihver1/android/yaph/IconCache.java:56]: magic number is always bad

File: src/ru/mihver1/android/yaph/ImagePage.java (hunk 5 above)
Comment (thevery) [inline src/ru/mihver1/android/yaph/ImagePage.java:49]: logs should be enabled in debug build only

**Pull request closed without merging**